cooltools/cli/compute_saddle.py

```diff
@@ -0,0 +1,515 @@
+    # validate expected a bit as well:
+    expected_chroms = get_exp_chroms(expected)
+    # do simple column-name validation for now:
+    if not set(track_chroms).issubset(expected_chroms):
+        raise ValueError("Chromosomes in {} must be subset of chromosomes in expected {}".format(track_path,
+                                                                                               expected_path))
+    # and again bins are supposed to match up:
+    # only for cis though ...
+    expected_bins = get_exp_bins(expected,track_chroms,track_bins)
+    if not (track_bins==expected_bins):
+        raise ValueError("Number of bins is not matching:",
+                " {} in {}, and {} in {} for chromosomes {}".format(track_bins,
+                                                                   track_path,
+                                                                   expected_bins,
+                                                                   expected_path,
+                                                                   track_chroms))
+    #############################################
+    # CROSS-VALIDATION IS COMPLETE.
+    #############################################
+    # COOLER, TRACK and EXPECTED seems cross-compatible:
+
+    # define OBS/EXP getter functions,
+    # it's contact_type dependent:
+    if contact_type == "cis":
+        obsexp_func = make_cis_obsexp_fetcher(c, expected, expected_name)
+    elif contact_type == "trans":
+        obsexp_func = make_trans_obsexp_fetcher(c, expected, expected_name)
+
+
+
+    # digitize the track (i.e., compartment track)
+    # no matter the contact_type ...
+    track_fetcher = make_track_fetcher(track, track_name)
+    track_mask_fetcher = make_track_mask_fetcher(track, track_name)
+    digitized, binedges = saddle.digitize_track(
+                                    bins = n_bins,
+                                    get_track = track_fetcher,
+                                    get_mask  = track_mask_fetcher,
+                                    chromosomes = track_chroms,
+                                    prange = prange,
+                                    by_percentile = by_percentile)
+    # digitized fetcher, yielding per chrom slice:
+    digitized_fetcher = lambda chrom: digitized[chrom]
+
+
+    # Aggregate contacts (implicit contact_type dependency):
+    sum_, count = saddle.make_saddle(
+        get_matrix = obsexp_func, 
+        get_digitized = digitized_fetcher,
+        chromosomes = track_chroms,
+        contact_type = contact_type,
+        verbose = verbose)
+    # actual saddleplot data:
+    saddledata = sum_/count
+
+    ##############################
+    # OUTPUT AND PLOTTING:
+    ##############################
+
+    # # output to stdout,
+    # # just like in diamond_insulation:
+    if output is not None:
+        # output saddledata (square matrix):
+        pd.DataFrame(saddledata).to_csv(output+".saddledata.tsv",
```

Comment:
This is fine. An alternative way to do the same is `np.savetxt`, but even less flexible. You could also output `.npy` if that extension is used instead.

Reply:
see reply for the next comment